src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1411,6 +1412,28 @@ public function testNoCollectDenormalizationErrorsWithWrongEnumOnConstructor()
         }
     }
 
+    /**
+     * @requires PHP 8.1
+     */
+    public function testNoCollectDenormalizationErrorsWhenObjectNotInstantiable()
+    {
+        $serializer = new Serializer(
+            [
+                new BackedEnumNormalizer(),
+                new ObjectNormalizer(),
+            ],
+            ['json' => new JsonEncoder()]
+        );
+
+        try {
+            $serializer->deserialize('{}', UnitEnumDummy::class, 'json', [
+                DenormalizerInterface::COLLECT_DENORMALIZATION_ERRORS => true,
+            ]);
+        } catch (\Throwable $th) {
+            $this->assertInstanceOf(NotNormalizableValueException::class, $th);
```

Comment:
```suggestion
        } catch (\Throwable $e) {
            $this->assertInstanceOf(NotNormalizableValueException::class, $e);
```

but this should use expectException instead

Reply:
Agreed it makes more sense to use expectException so updated the test 👍